### ISSUE TITLE
Ignore SC2317 on: return $STATUS 2>&- || exit $STATUS

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#2620]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2317
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
# shellcheck enable=all

STATUS=0
# Some code
return "${STATUS}" 2>&- || exit "${STATUS}"

```

#### Here's what shellcheck currently says:
[SC2317](https://www.shellcheck.net/wiki/SC2317) (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

#### Here's what I wanted or expected to see:
No message, since I want to support both being executed and being sourced.
`return` will apply when sourced while `exit` will apply when executed.

**Comment by akinomyoga** [comment]
We also hit this false positive in https://github.com/ohmybash/oh-my-bash/pull/402#discussion_r1111634730. The pattern `if <some test command>; then return xxx 2>/dev/null || exit xxx; fi` is an idiom for the scripts that can be used for both executables and source scripts. Because of this false positive information, users would try to remove `|| exit xxx` in PRs, which would break the script. This is harmful. Could anyone support the detection of this case and suppress the information?

**Comment by forthrin** [comment]
+1. Does anyone know a way to return / exit from a script regardless of sourcing or running? If not, it seems this should be fixed, though with a backlog of +1k issues, maybe unlikely.
